### ISSUE TITLE
[Bug] OpenMSX hangs. How to trigger this event? [sf#366]

### ISSUE THREAD

**Issue by openMSX-import** [openMSX/openMSX#760]
**Reported by hydragon on 2009-12-28 23:48 UTC**
In the video I demonstrate how to trigger the bug.
But I will explain it in this messag as well:
Start openMSX, go to Full Screen with [ALT] + [ENTER]
Everthing works fine from here, Game loads smooth.
** Let's generate the bug **
- Press [WinIcon] + [D] to return to Desktop.
- Results:
* Music hangs
* Mouse cursor remains in left top, any movement doesn't work.
- Pressing [ALT] + [TAB] to switch back to OpenMSX
- Screen hangs as well, you can press any key you want, but nothing will happen.
- [CTRL] + [ALT] + [DEL] and bring up task manager.
- You will notice in teh background the openMSX screen turned white! 
- Extreme CPU usage : 50%, and abou 120 Mb of Ram.
- End Task, and OpenMSX Exists.
- End of triggering and exiting OpenMSX.

/- Version : 0.7.2 [Dev. 11034]
/- Build: Visual Studio 2008
/- OS: Windows Vista Home Premium [32bit]
/- Videolink: http://www.youtube.com/watch?v=UORzWRoMH6A

Greetings, Hydragon

**Comment by openMSX-import** [comment]
**Commented by hydragon on 2009-12-29 00:50 UTC**
/- Version : 0.7.2 [Dev. 11023]
My bad!

**Comment by openMSX-import** [comment]
**Commented by mfeingol on 2010-02-13 08:29 UTC**
I can repro this on Win7 with the latest SVN.  I believe it's an SDL bug.

When the issue occurs, I see an endless stream of SDL_MOUSEMOTION events, even though the mouse isn't being touched.  This appears to block openMSX from making progress, because we never return from InputEventGenerator::poll().

**Comment by openMSX-import** [comment]
**Commented by manuelbi on 2010-02-13 09:02 UTC**
Is it worth to investigate how we can make a patch for SDL to fix or work around this problem?

**Comment by openMSX-import** [comment]
**Commented by mfeingol on 2010-02-15 20:09 UTC**
I took a look at what was going on inside SDL.  SDL thinks the mouse is moving, even though it isn't.  This occurs in DIB_GenerateMouseMotionEvent.  When the WIN+D (or ALT+TAB) key is pressed, the screen size reverts to the original size (instead of 1024x768) and Windows starts returning some rather odd values for the physical position of the middle of the window  (e.g. center.x = -31520, center.y = -31640).

Long story short, the workaround I applied was a patch to this SDL code to cause it to only post a new event if (after calling SetCursorPos) the cursor's position actually changed from the original position (it doesn't, it's stuck at 0,0).

Anyway, this appears to fix the hang - revision 11269.  Not that full-screen works all that well anyway, but at least it doesn't hang.

**Comment by openMSX-import** [comment]
**Updated by mfeingol on 2010-02-15 20:09 UTC**
- **status**: open --> closed-fixed

**Comment by openMSX-import** [comment]
**Commented by manuelbi on 2010-02-19 14:31 UTC**
Also in this case, wouldn't it be useful to try to push the patch upstream to SDL?

**Comment by openMSX-import** [comment]
**Commented by mfeingol on 2010-02-20 20:47 UTC**
I submitted an SDL bug:  http://bugzilla.libsdl.org/show\_bug.cgi?id=956
